src/main/java/io/jenkins/plugins/sprp/PipelineSnippetGenerator.java

```diff
@@ -99,19 +105,134 @@ else if(agent.getDockerfile() != null){
 
                 agentLines.add(getCommonOptionsOfAgent(agent));
                 agentLines.add("}");
+                agentLines.add("}");
             }
-            else {
+            else if(agent.getLabel() != null && agent.getCustomWorkspace() != null){
+                agentLines.add("agent {");
                 agentLines.add("node{");
                 agentLines.add(getCommonOptionsOfAgent(agent));
                 agentLines.add("}");
+                agentLines.add("}");
+            }
+            else {
+                agentLines.add("agent any");
             }
 
-            agentLines.add("}");
         }
 
+        if(agent != null)
+            agentLines.addAll(getTools(agent.getTools()));
```

Comment:
Please use [1TBS](https://en.wikipedia.org/wiki/Indentation_style#Variant:_1TBS_(OTBS)). In other words, put braces whether the compiler needs them or not. Do this everywhere, not just here.